dev-tool/src/list_net/data_source.rs

```diff
@@ -319,18 +335,50 @@ impl InMemorySamples {
         })
     }
 
-    /// Add a new sample.
+    /// Prepares samples to be added to this storage.
+    ///
+    /// This is split out from adding samples so that the preparation
```

Comment:
What do you mean by `split out`?

Reply:
I will update the documentation, originally this was done "on-the-fly" when adding samples, but for supporting parallelization it's easier to do it in a separate step and then add it.  (Also allows of testing both things independently.)